### PR TITLE
CB-14021 - fix polling upon successful experience deletion

### PR DESCRIPTION
**Pull request by gregito**

### CODE REVIEW

**Review by doktoric — APPROVED** [review]

File: environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentExperienceDto.java [hunk 1]
```diff
@@ -80,6 +90,14 @@ public EnvironmentExperienceDto build() {
             return new EnvironmentExperienceDto(name, crn, accountId, cloudPlatform);
         }
 
+        public EnvironmentExperienceDto fromEnvironment(Environment environment) {
```
Comment (bergerdenes) [inline environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentExperienceDto.java:93]: is this function needed?

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/experience/EnvironmentExperienceDeletionAction.java [hunk 2]
```diff
@@ -84,7 +87,14 @@ private void waitForResult(Environment environment, boolean forceDelete) {
...
         if (!isSuccess(result.getLeft())) {
-            processFailureWithForceDelete(result, forceDelete);
+            if (result.getLeft().equals(EXIT)) {
+                reCheckDeletionOtherwise(
+                        environment,
+                        env -> experienceConnectorService.getConnectedExperienceCount(EnvironmentExperienceDto.fromEnvironment(env)) == 0,
+                        () -> processFailureWithForceDelete(result, forceDelete));
+            } else {
+                processFailureWithForceDelete(result, forceDelete);
+            }
```
Comment (bergerdenes) [inline environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/experience/EnvironmentExperienceDeletionAction.java:97]: test?

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentExperienceDto.java (hunk 1 above)
Comment (gregito) [reply environment/src/main/java/com/sequenceiq/environment/environment/dto/EnvironmentExperienceDto.java:89]: now it is. forgot to call :)

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/experience/EnvironmentExperienceDeletionAction.java (hunk 2 above)
Comment (gregito) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/experience/EnvironmentExperienceDeletionAction.java:97]: done. pls check if it is suffucient

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**